### PR TITLE
feat(database): support using arrow keys to select kanban cells

### PR DESCRIPTION
**Pull request by zqran**
part of: #4494 

https://github.com/toeverything/blocksuite/assets/15389209/c57645af-520a-45bd-80f7-33ab634e6e64

### CODE REVIEW

**Comment by Saul-Mirone** [comment]
Just a hint: we have [command](https://github.com/toeverything/blocksuite/pull/4540) now so you can reuse some duplicate logic with it.

File: packages/blocks/src/database-block/kanban/selection.ts [hunk 1]
```diff
@@ -120,33 +121,260 @@ export class KanbanSelection {
...
+    );
+    const cardPos = nextCards
+      .map((card, index) => {
+        const targetRect = card.getBoundingClientRect();
+        return {
+          offsetY: getYOffset(rect, targetRect),
+          index,
+        };
+      })
+      .reduce((prev, curr) => {
```
Comment (zzj3720) [inline packages/blocks/src/database-block/kanban/selection.ts:204]: Does this cause an error if there is only one cell?

File: packages/blocks/src/database-block/kanban/selection.ts [hunk 2]
```diff
@@ -120,33 +121,260 @@ export class KanbanSelection {
...
+        if (prev.offsetY < curr.offsetY) {
+          return prev;
+        }
+        return curr;
+      });
+    const nextCell = cells[cellPos.index];
+
+    return {
+      cell: nextCell,
+      cardId: nextCard.getAttribute('data-card-id') as string,
```
Comment (zzj3720) [inline packages/blocks/src/database-block/kanban/selection.ts:233]: Maybe nextCard.cardId would be more typesafe.

**Review by zzj3720 — APPROVED** [review]

**Comment by Saul-Mirone** [comment]
We should add some tests for this feature.

**Review by doodlewind — APPROVED** [review]
Should be good to have tests in following PRs.

**Pull request merged**